dask_ml/model_selection/_successive_halving.py

```diff
@@ -0,0 +1,239 @@
+import math
+
+import numpy as np
+import toolz
+
+from ._incremental import IncrementalSearchCV
+
+
+class SuccessiveHalvingSearchCV(IncrementalSearchCV):
+    """
+    Perform the successive halving algorithm.
```

Comment:
A bit more here would be appreciated, and a ref to the standard academic paper on successive halving.